docs/charter.md

```diff
@@ -1,28 +1,70 @@
 ICU4X Charter
 ==============
 
-ICU4X is a new project whose objective is to solve the needs of clients who wish to provide client-side i18n for their products in resource-constrained environments.
+The ICU4X project designs and develops a set of modular internationalization components suitable for use on the client-side and in resource-constrained environments.
 
-ICU4X will be built from the start with several key design constraints:
+ICU4X builds on the design and API decisions of [`ICU4C`]/[`ICU4J`] and [`ECMA-402`].
```

Comment:
Optional: Using backticks seems like an odd style choice.  I would just write `[ICU4C]` (and ICU4J, ECMA-402).  They will turn to blue links, which I think is enough to highlight them from the background.